### PR TITLE
Fix readBits & Add some feature

### PR DESCRIPTION
**Pull request by Lee20171010**
1. Fix readBits function error calculate the end of the file
2. Add BigInt support, since if we need to calculate some number, for example adding them, the result may be huge
3. Add function parameter to display Offset with hex value, since if used to parse memory, hex offset is more intuitive
4. Add a dropdown for users to select the parser they want

### CODE REVIEW

**Comment by maziac** [comment]
Thanks for the contribution.
Please allow some time for me to evaluate.

File: src/html/dataread.ts
Comment (maziac) [inline src/html/dataread.ts:1]: OK

**Comment by maziac** [comment]
1. Fix readBits function error calculate of end of file
OK 

2. Add BigInt support
OK

3. Add function parameter to display Offset with hex value
I wouldn't want to change the API for this feature. If at all I guess it makes more sense to change it globally then for each row.
What do you think, could you omit the new hexOffset parameter and the change to relHexOffset variable and always display both, dec and hex, on hover?

4. Add dropdown for user to select the parser they want
I think I would need some explanation on this. I tried it but for me the select box always showed only one file.

**Comment by Lee20171010** [comment]
3. Do you mean displaying the Offset in hex by default?
I’m concerned that this might just be my personal preference, so I think it’s better to keep the current implementation. I’ll revert it and only maintain this change in my fork.

4. I added this feature because when multiple parsers are in the filepath, the current implementation only shows a warning and uses the first parser. However, if I have many different binary files that require different parsers, I would need to keep changing the parsing path. Therefore, I added a dropdown to allow users to choose which parser they want to use for parsing each binary file.

Example:
Currently having 2 parsers in a filepath, naming tst1.js & tst2.js

when I open a binary file, it will be using the 1st parser by default
image

but I can choose a different parser in the dropdown option
image

then press reload to parse this file using the different parser
image

I believe users might have different files that require parsing with various parsers, so it would be beneficial to let them choose which parser to use. However, I’m not very familiar with VSCode development and TypeScript, so my suggested implementation might not be the best approach.

**Comment by maziac** [comment]
3. I meant: Mainly leave as it is but display hex and decimal offset on hover.
4. I See. I have a look once more.

**Comment by maziac** [comment]
3. We could also put the new argument as last argument. So it would not break the any existing parsers for the users.

**Comment by maziac** [comment]
4. The changes are fine for me. But could you mention the behavior somewhere in the docs.

**Comment by maziac** [comment]
3. Maybe we could smile change the order of the arguments, so that the new one is the last.

**Comment by Lee20171010** [comment]
Sorry for the late reply.

3. Changing the order of the arguments is a good idea~
4. OK, I will add this to the walkthrough and show help.

I plan to modify(3,4) and push a new commit this weekend. I would appreciate it if you could review it for me when the time comes. Thank you!

**Comment by maziac** [comment]
Looks good to me.
Only thing, the (3) function signature is not documented yet. But I could also do it myself (then this comment serves  as a reminder).

**Comment by Lee20171010** [comment]
Hi, about the function signature, does it mean the following thing?
image

**Comment by maziac** [comment]
Yes

**Comment by maziac** [comment]
Thanks for your contribution. (And your patience.)

**Comment by maziac** [comment]
Just released 1.11.0 with the changes.

**Pull request merged**